### DISCUSSION TITLE
Does current C api support creating nested logical type?

### DISCUSSION THREAD

**Discussion by metab0t** [duckdb/duckdb#3452] [Q&A] [answered]
I only find `duckdb_create_logical_type` with one argument.

Is it possible to create nested complex logical type via C api?

**Comment by Mytherin** [comment, marked as answer]
It is not possible to create nested types right now, but this should be relatively straightforward to implement. Happy to review a PR!

**Reply by metab0t** [reply]
I have opened a PR https://github.com/duckdb/duckdb/pull/3456

**Comment by Mause** [comment]
I believe this is now implemented for lists and structs

**Reply by Giorgi** [reply]
I saw that your PR was merged but to be honest I'm not sure how to create `duckdb_value`-s that I need to pass to the new functions you added. I only see `duckdb_create_varchar` and `duckdb_create_int64` in the API docs. How do I create a list of ints, a list of hugeint, or a list of timestamps?

**Reply by Giorgi** [reply]
@Mause ?

**Reply by Mause** [reply]
> @Mause ?

Yep, I'm looking at that now
